### PR TITLE
Use better variable names and remove unwanted code blocks.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: hchiramm <hchiramm@redhat.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/cloudprovider/providers/photon/photon.go [hunk 1]
```diff
@@ -474,16 +474,16 @@ func (pc *PCCloud) InstanceID(ctx context.Context, nodeName k8stypes.NodeName) (
 	name := string(nodeName)
 	if name == pc.localK8sHostname {
 		return pc.localInstanceID, nil
+	}
+	// We assume only master need to get InstanceID of a node other than itself
+	ID, err := getInstanceID(pc, name)
+	if err != nil {
```
Comment (liggitt) [inline pkg/cloudprovider/providers/photon/photon.go]: simplify to this?
```
if err != nil {
  klog.Errorf("Photon Cloud Provider: getInstanceID failed for InstanceID. Error[%v]", err)
}
return ID, err
```

**Review by liggitt — COMMENTED** [review]
a couple nits. given https://github.com/kubernetes/kubernetes/pull/72178, do we are about little cleanups to this code?

File: pkg/cloudprovider/providers/photon/photon.go [hunk 2]
```diff
@@ -474,16 +474,16 @@ func (pc *PCCloud) InstanceID(ctx context.Context, nodeName k8stypes.NodeName) (
 	name := string(nodeName)
 	if name == pc.localK8sHostname {
 		return pc.localInstanceID, nil
+	}
+	// We assume only master need to get InstanceID of a node other than itself
+	ID, err := getInstanceID(pc, name)
```
[the remark this answers is not in the document]
Comment (humblec) [reply pkg/cloudprovider/providers/photon/photon.go]: fine to me. Done.

File: pkg/cloudprovider/providers/photon/photon.go (hunk 1 above)
Comment (humblec) [reply pkg/cloudprovider/providers/photon/photon.go]: sgtm. Done.

File: pkg/cloudprovider/providers/photon/photon.go [hunk 3]
```diff
@@ -544,7 +544,7 @@ func (pc *PCCloud) HasClusterID() bool {
 	return true
 }
 
-// Attaches given virtual disk volume to the compute running kubelet.
+// AttacheDisk attaches given virtual disk volume to the compute running kubelet.
```
[the remark this answers is not in the document]
Comment (humblec) [reply pkg/cloudprovider/providers/photon/photon.go]: done.

**Comment by humblec** [comment]
@liggitt thanks , addressed the comments. Yeah, this may be deprecated, but its going to take some time, till then its good to have it addressed in our source.

**Pull request merged**